### PR TITLE
[dashboard] show repo name when workspace creation fails

### LINKED ISSUE #7191: Display name of the repository on the "Repository not found" error screen
## Describe the behaviour you'd like

I made a mistake with the URL passed into the context parser. 

Gitpod provided a button to try again

So I mashed the try again button

Turns out it's user error but that could have been clearer!

Browsers now hide URLs!

### PR DESCRIPTION
**Pull request by trumbitta**
## Description

When the creation of a workspace fails because the repo wasn't found, also display the repository name near the error messages so that in the scenario described by #7191 it becomes easier to understand what's actually going on.

## Related Issue(s)

Fixes #7191 

## How to test

- Run the dashboard app
- Try to create a workspace from a non-existent repository. E.g. https://3000-teal-earthworm-ch85f047.ws-eu23.gitpod.io/#https://github.com/gitpod-io/gitpoddenz
- Look for the updated error message (See screenshot below)

image

## Release Notes

```release-note
[dashboard] Error messages on workspace creation when the repository is not found, will now also display the name of the repository
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

Associated issue: *#7191*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by trumbitta** [comment]
Great catch!

I'd rather have the message **never** include the repo name though: I find it to be clearer in the new position added by this PR, and it also makes it easier to use a `<code>` HTML tag which is the semantically appropriate way to include a repo name in the document 🤔

**Comment by trumbitta** [comment]
> Sometimes, the message returned from the server already carries the repo name My suggestion is to adjust the server so that all messages returned contain the repo name, which is better than adjusting the dashboard

Wait a minute, hold your horses: it seems we both have been misguided and the actual user-facing messages are in the frontend component (as they should!).

I'll work on an update asap!

**Comment by trumbitta** [comment]
@iQQBot ok, done. I also added a commit with a quick semantic improvement for the repo owner in the messages 🍻

**Comment by trumbitta** [comment]
Please good bot, keep it open 🙏

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (Siddhant-K-code) [inline components/dashboard/src/start/CreateWorkspace.tsx:262]:
```suggestion
          <p className="text-base text-gray-400 w-96">The repository <code>${repoName}</code> may be private. Please authorize Gitpod to access to private repositories.</p>
```

**Review by Siddhant-K-code — COMMENTED** [review]
I think we should return the Repo. Name too.

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx:262]: Care to elaborate on why?

I reckon that keeping it inside the error message makes it pass as "unimportant" and makes also easily misread.
In the new proposed position the repo name is front and center, bigger, red: meaning that next time, the next user won't likely have the same hiccup @ghuntley lamented about in #7191 😄

Comment (Siddhant-K-code) [reply components/dashboard/src/start/CreateWorkspace.tsx:262]: Just a more informative way to display to the users, maybe!?

Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx:262]: I forgot to mention that keeping the repo name also in the error message looked bad both to me and @iQQBot . See https://github.com/gitpod-io/gitpod/pull/7302#issuecomment-997430263

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (gtsiolis) [inline components/dashboard/src/start/CreateWorkspace.tsx:294]: **thought**: We could probably use the same `CodeText` component here as mentioned below. 💭

**Review by gtsiolis — APPROVED** [review]
Thanks @trumbitta for contributing, @Siddhant-K-code for chiming in, and @iQQBot for doing a first review round! 🏀 

@trumbitta Changes here definitely improve overall user experience of this page. 🔮 

Left some minor comments below. Let me know what you think but we could also leave these out of the scope of this PR. Your call.

Approving to unblock merging but holding in case we could tackle some minor UX issues as mentioned below.

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 0b9c1446fad28347ffbf946cf7a256a47755a85a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7302#pullrequestreview-847622451" title="Approved">gtsiolis</a>*

Associated issue: *#7191*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by Siddhant-K-code** [comment]
Literally one of the best & informative code review by @gtsiolis ✨

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx:303]: Thanks! I didn't know about `CodeText`, and it looked promising at first... but it's a `span` whereas I would rather use a `code` for this.

What's the intended use of `CodeText`? Why would I want to have a styled `span` that looks like a styled `code` instead of using a styled `code`?
For example, those variable names in `Account.tsx` should be marked up with a `code`.

So now I need advice on how to handle this in the best interest of the project 😊:

- [ ] Use `CodeText` as-is (🙈 I'd rather not contribute more non-semantic markup)
- [ ] Leave `code` here but copy-paste the style from `CodeText` (🤕 repetition over encapsulation)
- [ ] Modify `CodeText` to be a `code` instead of a `span` (🤔 but are there any cases in the codebase where a span is actually the right element?)
- [ ] Make `CodeText` into a polymorphic component via a classic `as` property, defaulting to `span` for retro-compatibility (🖌 might be too much, given it's unlikely anyone would ever want `CodeText` to render as a `section`, or a `dl`, or anything other than a `span` or a `code`.

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx:288]: Ok, pushing in a minute 👍

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (gtsiolis) [reply components/dashboard/src/start/CreateWorkspace.tsx]: Resolved in 653f02e.

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (gtsiolis) [reply components/dashboard/src/start/CreateWorkspace.tsx]: Resolved in 653f02e.

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (gtsiolis) [reply components/dashboard/src/start/CreateWorkspace.tsx:303]: 🍊 **Answering some questions:**

> Why would I want to have a styled span that looks like a styled code instead of using a styled code?

@trumbitta Using `<code>` tag to render such elements sounds right, especially since this is broadly related to rendering markdown elements like _code blocks_ and _inline code_. The idea behind `CodeText` component was to wrap such elements and provide proper styling when rendering instances across the dashboard.

> For example, those variable names in `Account.tsx` should be marked up with a code.

@trumbitta You're right and this can also help with accessibility and component consistency. I'm also open for discussing a different design between inline code like the variables in `/account` and the repository full name in the error pages here. 💯 

**🍋  Some background context:**

FWIW, for historical reasons, the `CodeText` component was introduced as a side fix during a larger feature implementation in https://github.com/gitpod-io/gitpod/pull/4607, to replace the already existing span-based elements from the initial dashboard redesign. To help cut some corners and unblock shipping that larger feature, we had to limit [scope creep](https://en.wikipedia.org/wiki/Scope_creep) by limiting improvements related to these code elements to just componetizing the already existing elements. 🔖 

FYI, we're approaching the development of the dashboard using an MVC (Minimum Viable Change) approach and we iteratively improve existing or add new functionalities. ➿ 

**🥝 Proposal:**

1. Taking all the above into account, in the spirit of MVC and iterating step-by-step as product needs grow, I'd suggest to keep the `CodeText` component as is for now. If you could use `CodeText` for replacing the repository full name below that would be great. Otherwise, feel free to leave this as is.
2. Next steps could include improving this `CodeText` component by using the `<code>` tag under the hood instead of a `<span>`, while checking if this renders ok across the dashboard.

Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx:303]: 💯 Makes sense.
Thanks for the bit of history and context explanation: now I feel more aligned with the approach and guiding principles 🎛️ . It will be easier to contribute and think strategically 🧐 !

Balancing MVC (first time I see this acronym with a glorious past used with this meaning 😸 ) with my old core principle of continuous refactoring (here's a fun take by Kent Beck: https://docs.google.com/document/d/e/2PACX-1vRS232prL-Bq44hmmdzJTHkCz-nYBlQVPAtemOmVfDP1rbiD777jQAaUpIeF2KCdSSEJMEwHTeo9QdE/pub) promises to be a refreshing and positive challenge 🙇 so thank you for that 🙏 

🪡 I'll update the bits you pointed out to use `CodeText`, and I will open an issue for improving it with the `<code>` tag.

Comment (Siddhant-K-code) [reply components/dashboard/src/start/CreateWorkspace.tsx:303]: Thanks a lot again @gtsiolis for the detailed explanation! ❤️

Comment (gtsiolis) [reply components/dashboard/src/start/CreateWorkspace.tsx:303]: Thanks @Siddhant-K-code @trumbitta! 🙏 💯

Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx]: @gtsiolis done!

**Comment by gtsiolis** [comment]
UX looks great, @trumbitta! 🔮 

@jankeromnes could you take another look at the code changes here when you're free?

**Comment by trumbitta** [comment]
oh come on stale bot, you old rascal!

**Comment by jankeromnes** [comment]
Whoops, looks like I've overseen this review request. Will take a look now! Many thanks 🙏

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (jankeromnes) [inline components/dashboard/src/start/CreateWorkspace.tsx:20]: Interesting, I had no idea we had this component. 😅 I agree with @trumbitta that [a `<code>` element would be more semantic than a `<span>`](https://github.com/gitpod-io/gitpod/pull/7302#discussion_r781160513).

Also, creating single-line React components is not the best way to share common style (because it adds unnecessary code and complexity, and it's too easy to forget about using the component where relevant). I think we should remove the `CodeText` component again, and instead apply our common CSS rules to the `code` element inside [index.css](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/index.css).

However, this is off-topic here and can definitely be a follow-up issue / Pull Request. @trumbitta would you be interested in filing an issue for this? 🙏

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (jankeromnes) [inline components/dashboard/src/start/CreateWorkspace.tsx:264]: **Praise**: Removing the `secondary` here is such an awesome drive-by fix. 💯 Many thanks @trumbitta and @gtsiolis. ❤️

**Review by jankeromnes — APPROVED** [review]
Also looks good to me! 🚀

Many thanks @trumbitta for this improvement; @gtsiolis for the awesome reviews; and @iQQBot and @Siddhant-K-code for helping out! 🙏

Left a few thoughts in-line, but they're not blocking so removing the hold.

Also, the preview environment got cleaned up so I couldn't test this, but since @gtsiolis already [approved the UX](https://github.com/gitpod-io/gitpod/pull/7302#issuecomment-1011383526) I don't need to. 🚢

Finally, I feel like adding the repository name inside the top error message would be even better (i.e. `Repository gitpod-io/gitpoddenz not found.` as opposed to `Repository not found.<br/>gitpod-io/gitpoddenz`) but looking at the code this looks trickier than anticipated, and the current fix looks good to me too.

/unhold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 477dc1647bc12f1407f5fdef8c227096f280f7d8</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7302#pullrequestreview-847622451" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7302#pullrequestreview-860684979" title="Approved">jankeromnes</a>*

Associated issue: *#7191*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/start/CreateWorkspace.tsx
Comment (trumbitta) [reply components/dashboard/src/start/CreateWorkspace.tsx:20]: I had already created one :) might be worth it to chime in and add your perspective? https://github.com/gitpod-io/gitpod/issues/7555

**Pull request merged**
